node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -556,6 +583,9 @@ impl<C: Context> ActiveValidator<C> {
                 endorsements.endorsers.iter().any(is_ours)
                     && !state.has_endorsement(endorsements.unit(), self.vidx)
             }
+            Vertex::Ping(ping) => {
+                ping.creator() == self.vidx && !state.has_ping(self.vidx, ping.timestamp())
```

Comment:
Why do we have to check if we have the ping message in the state? Is it being gossiped?

Reply:
Yes, it is.
`has_ping` returns `true` even if we have a ping or unit that's only slightly older, to prevent spam. So: If you send me two pings with less than one maximum round length in between, I won't gossip the second one.
And honest nodes will only produce a ping if their most recent ping or unit is old enough.

`has_ping` is a misnomer… maybe I should call it `needs_ping`, and negate it?

Reply:
That's fine, `has_ping` sounds good to me.